src/world.rs

```diff
@@ -669,14 +645,56 @@ impl EntityMetadata {
 }
 
 #[allow(clippy::missing_safety_doc)]
-pub unsafe trait Bundle: 'static {
+pub unsafe trait Bundle
+where
+    Self: 'static,
+{
+    fn contains<T>() -> bool
+    where
+        T: 'static;
+
     fn insert(types: &mut TypeMetadataSet);
     #[must_use]
     fn remove(types: &mut TypeMetadataSet) -> Option<()>;
+
+    unsafe fn drop<T>(archetype: &mut Archetype, idx: u32)
+    where
+        T: Bundle;
+
     unsafe fn write(self, archetype: &mut Archetype, idx: u32);
     unsafe fn read(archetype: &mut Archetype, idx: u32) -> Self;
 }
 
+struct Empty;
```

Comment:
This is private instead of extending the tuple impl to include `()` to avoid having to handle the empty bundle in `World::remove`.